cabal-install/src/Distribution/Client/Setup.hs

```diff
@@ -497,10 +497,16 @@ filterConfigureFlags flags cabalLibVersion
       configConstraints = []
     }
 
-    flags_3_11_0 = flags_latest {
-        configDependencies = configDependencies flags ++ configPromisedDependencies flags
-        , configPromisedDependencies = []
-    }
+    flags_3_11_0 = if not null configPromisedDependencies then error $
```

Comment:
Tried to overlay on a project but had to remove this commit to get it to build. Is there a `flags` and some parenthesis missing?

Reply:
@fgaz: that's probably why the CI fails?

Reply:
Yes I know, I'll unmark draft when I finish. When I pushed I was still building dependencies so I didn't test. I'll push something that build in a minute